W3ChampionsStatisticService/Matches/MatchRepository.cs

```diff
@@ -146,6 +144,15 @@ private PlayerScore CreateDetail(PlayerBlizzard playerBlizzard)
             return events;
         }
 
+        public Task<long> Count(
+            GateWay gateWay = GateWay.Undefined,
```

Comment:
I may misunderstand the full purpose, but do you think **Gateway.Any** or **GameMode.Any** would be more clear in the code than undefined?